hydromt_wflow/workflows/soilparams.py

```diff
@@ -163,3 +163,79 @@ def update_kvfrac(
             da_kvfrac.loc[dict(layer=idx)] = kvfrac
 
     return da_kvfrac
+
+
+def get_ks_veg(KsatVer, sndppt, LAI, alfa=4.5, beta=5):
+    """
+    based on Bonetti et al. (2021) https://www.nature.com/articles/s43247-021-00180-0.
+
+    Parameters
+    ----------
+    KsatVer : [xr.DataSet, float]
+        saturated hydraulic conductivity from PTF based on soil properties [cm/d].
+    sndppt : [xr.DataSet, float]
+        percentage sand [%].
+    LAI : [xr.DataSet, float]
+        Mean annual leaf area index [-].
+    alfa : float, optional
+        Shape parameter. The default is 4.5 when using LAI.
+    beta : float, optional
+        Shape parameter. The default is 5 when using LAI.
+
+    Returns
+    -------
+    ks : [xr.DataSet, float]
+        saturated hydraulic conductivity based on soil and vegetation [cm/d].
+
+    """
+    # get the saturated hydraulic conductivity with fully developed vegetation.
+    ksmax = 10 ** (3.5 - 1.5 * sndppt**0.13 + np.log10(KsatVer))
+    # get the saturated hydraulic conductivity based on soil and vegetation mean LAI
+    ks = ksmax - (ksmax - KsatVer) / (1 + (LAI / alfa) ** beta)
+    return ks
+
+
+def ksatver_vegetation(self, KsatVer, sndppt, LAI, alfa=4.5, beta=5):
```

Comment:
Instead of self here, you can decide to pass ds_like (self.grid) and in the doc of the function say the required variables are KsatVer, LAI and wflow_subcatch. Or keep it as three separate argument (add one for the mask)